app/components/Contacts/ContactFormRefactor/ContactFormRefactor.jsx

```diff
@@ -0,0 +1,311 @@
+// @flow
+import React from 'react'
+import { intlShape } from 'react-intl'
+import { wallet } from '@cityofzion/neon-js'
+import { wallet as n3Wallet } from '@cityofzion/neon-js-next'
+import { Box } from '@chakra-ui/react'
+import { BSNeo3 } from '@cityofzion/bs-neo3'
+
+import Button from '../../Button'
+import TextInput from '../../Inputs/TextInput'
+import AddContactIcon from '../../../assets/icons/contacts-add.svg'
+import styles from './ContactFormRefactor.scss'
+import AddIcon from '../../../assets/icons/add.svg'
+import TrashCanIcon from '../../../assets/icons/delete.svg'
+import { useContactsContext } from '../../../context/contacts/ContactsContext'
+import { ROUTES } from '../../../core/constants'
+
+type Props = {
+  name: string,
+  submitLabel: string,
+  intl: intlShape,
+  showSuccessNotification: ({ message: string }) => any,
+  formAddress?: string,
+  history: {
+    push: Function,
+  },
+  handleSubmit?: () => void,
+}
+
+export default function ContactForm(props: Props) {
+  const {
+    intl,
+    submitLabel,
+    showSuccessNotification,
+    history,
+    formAddress,
+  } = props
+  const { contacts, updateContacts, deleteContact } = useContactsContext()
+  const [addressCount, setAddressCount] = React.useState(1)
+  const [errorMapping, setErrorMapping] = React.useState({
+    addresses: [],
+    name: '',
+  })
+  const [name, setName] = React.useState('')
+  const [addresses, setAddresses] = React.useState([formAddress || ''])
+  const [loading, setLoading] = React.useState(false)
+
+  // this indicates we are in edit mode
+  React.useEffect(
+    () => {
+      if (props.name && contacts[props.name]) {
+        setName(props.name)
+        setAddresses(
+          contacts[props.name] &&
+            contacts[props.name].map(contact => contact.address),
+        )
+        setAddressCount(contacts[props.name]?.length)
+      }
+    },
+    [props.name],
+  )
+
+  function handleChangeName(event) {
+    const existingNames = Object.keys(contacts)
+
+    if (existingNames.includes(event.target.value)) {
+      setErrorMapping({
+        ...errorMapping,
+        name: intl.formatMessage({ id: 'errors.contact.nameDupe' }),
+      })
+    }
+    setName(event.target.value)
+  }
+
+  async function handleChangeAddress(event, index) {
```

Comment:
This method is very big, do you think it's interesting to break it in smaller methods? Specially to be easier to have a general understanding on the multiple steps inside it

Reply:
Broke things down into a couple different functions, thanks again was just focused on getting the feature wrapped up